### PR TITLE
Camera switching in a dropdown list

### PR DESCRIPTION
**Pull request by waridrox**
Camera switching workflow for video stream: 

- Get all the possible devices in a dropdown
- For each `deviceId` associate a video stream 
- On device change through dropdown, stop the current stream and switch to the newly selected stream

Hi @jywarren, the current implementation is such that it takes the initial `facingMode` as `undefined`, since the devices have not been fetched yet. 

```javascript

 var constraints = {
        video: {
          deviceId: videoSource ? {exact: videoSource} : undefined //Taking device ids as the video source 
        }
      };

```

Even after fetching, it is unclear as to what device is facing the `environment` since some devices don't actually have a facingMode as `environment`. The MDN docs recommend using it for [smartphones](https://developer.mozilla.org/en-US/docs/Web/API/MediaTrackConstraints/facingMode). 
For instance if a laptop is connected to an external camera, that external camera is recognised as a `facingMode = environment` instead of the primary one.

Thus, I wanted to know your thoughts regarding this...

Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- see README.md for how to run them
* [ ] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [ ] pull request are descriptively named
* [ ] if possible, multiple commits squashed if they're smaller changes
* [ ] reviewed/confirmed/tested by another contributor or maintainer

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

**Comment by jywarren** [comment]
Hmm. What's the specific problem we're trying to fix in this PR? If we aren't always sure of the initial state, let's just say it's OK whatever the default is (after we give out best guess preference) and then if it's wrong the user will choose the better option from the dropdown. Would that work, or am I not quite getting your question? Thank you!!

**Comment by waridrox** [comment]
Sorry for the confusion 😅 I meant to say that when I tried out with wifi cam on my system, the primary cam was being identified as the wifi cam instead of the usual laptop cam

So there isn't a proper way to determine if a device even has a facingMode = environment in the first place. Because then the laptop cam should've been the environment facing one

> let's just say it's OK whatever the default is (after we give out best guess preference) and then if it's wrong the user will choose the better option from the dropdown. Would that work

Yes that would work 💯

File: dist/capture.dist.js [hunk 1]
```diff
@@ -265,20 +265,98 @@ $W.getUserMedia = function(options) {
...
+      if (typeof Promise === 'undefined') {
+      navigator.getUserMedia(constraints, successCallback, function(){}); 
+      } 
+      else {
+      navigator.mediaDevices.getUserMedia(constraints)
+      .then(successCallback);
+      } 
+    }
+  
+    videoSelect.onchange = start; //repeating the process for source change
```
Comment (jywarren) [inline dist/capture.dist.js:357]: It's saying `Cannot set property 'onchange' of null` probably in relation to this line -- do we have to modify anything for the cypress tests to see the `videoSelect` element?

Comment (waridrox) [reply dist/capture.dist.js:357]: I think there was no `dropdown` element for the `camera selection` in the interface, that's why the tests were failing 😅

**Comment by jywarren** [comment]
🎉🎉

**Comment by jywarren** [comment]
Should we bump the version and release a 0.0.x update?

**Comment by jywarren** [comment]
And, if you can link all the issues/PRs together as we do the publication process (including to this one) that would be great.

**Comment by waridrox** [comment]
Sure as far this version bump, there are these PRs only

_Changelog_ for v0.2.1 

#247 - Allows changing of video device streams in the form of a dropdown.
#245 - Allows to download the live spectrum image in the form of a `.png` file.

**Comment by jywarren** [comment]
Sounds great. Would you like to open a PR for the updated version or shall i do it when I publish? Either way works for me!

**Comment by jywarren** [comment]
I went ahead and bumped it and published!! 

https://github.com/publiclab/spectral-workbench.js/commit/3e93995486070f865db23834a81e34f81709d7cd

https://www.npmjs.com/package/spectral-workbench

**Comment by waridrox** [comment]
🎉 🚀 Thanks @jywarren!!! Though I think the dropdown functionality was still on hold...😅

**Comment by jywarren** [comment]
Ah, well we can release it in the next publication! 

Sorry, I took your 👍  as meaning we can go ahead. I'll be more careful to get an explicit "yes" in the future, thanks!!!

**Comment by jywarren** [comment]
I merged a bunch of dependabot PRs at https://github.com/publiclab/spectral-workbench/pulls and we should see this one reach there soon. If you want we can resolve the dropdown one and then merge after, but it's not a big deal to go through the process twice!

**Comment by waridrox** [comment]
No it's just that I included the dropdown element on the interface at `/v2` so that's why I was thinking about the dropdown one 😅

**Comment by jywarren** [comment]
Ah, ok! let's merge that too then!

**Comment by jywarren** [comment]
Ah, no, ok - so i guess actually the dropdown one is all good then, as we merged https://github.com/publiclab/spectral-workbench.js/pull/247 ?

**Comment by waridrox** [comment]
Yes it is 🙌

**Comment by jywarren** [comment]
Ah, sorry, maybe I still misunderstand. Is there something missing we need to fix now, or are we OK to take this a little slower? 

Ah, i see your response that it's OK now. Thanks!

Let's try to document clearly the interdependencies between PRs so we can sequence them correctly. Let's use clear comments like `Before merging this` or `Immediately after merging this...` etc!

Thanks!!!

**Comment by waridrox** [comment]
So the new interface also has a dropdown button which would be non-functional if we don't merge #245. The camera functionality was crucial so that is already done.

**Pull request merged**